spec/requests/public_spec.rb

```diff
@@ -2,156 +2,184 @@
-        expected_response = {
-          'Access-Control-Allow-Origin' => 'http://192.168.0.10:8080',
-          'Access-Control-Expose-Headers' => expose_headers,
-          'Access-Control-Allow-Credentials' => 'true',
-          'Access-Control-Allow-Methods' => allow_methods,
-          'Access-Control-Max-Age' => '7200'
-        }
-        expect_headers_to_include(expected_response)
-      end
-    end
-
-    context 'is invalid' do
-      example 'without Origin' do
-        headers = {
-          'Access-Control-Request-Method' => 'Origin, Content-Type, Accept, Authorization, Token',
-          'Access-Control-Request-Headers' => 'PUT',
-          'Accept' => 'application/json'
-        }
-        options '/projects', headers: headers
-
-        expect(response).to have_http_status(:bad_request)
-        expect_error(400, "The request was not valid: CORS preflight request to 'projects' was not valid. Required headers: Origin, Access-Control-Request-Method. Optional headers: Access-Control-Request-Headers.")
-      end
-
-      example 'without Access-Control-Request-Method' do
-        headers = {
-          'Origin' => 'http://localhost:3000',
-          'Access-Control-Request-Headers' => 'Origin, Content-Type, Accept, Authorization, Token',
-          'Accept' => 'application/json'
-        }
-        options '/projects', headers: headers
-
-        expect(response).to have_http_status(:bad_request)
-        expect_error(400, "The request was not valid: CORS preflight request to 'projects' was not valid. Required headers: Origin, Access-Control-Request-Method. Optional headers: Access-Control-Request-Headers.")
-      end
-
-      example 'with Origin that is not allowed' do
-        headers = {
-          'Origin' => 'http://localhost-not-allowed:3000',
-          'Access-Control-Request-Method' => 'GET',
-          'Access-Control-Request-Headers' => 'Origin, Content-Type, Accept, Authorization, Token'
-        }
-
-        options '/projects', headers: headers
-        expect_empty_body
-        expect(response).to have_http_status(:ok)
-        expected_response = {
-          'Access-Control-Allow-Origin' => 'http://localhost-not-allowed:3000',
-          'Access-Control-Expose-Headers' => expose_headers,
-          'Access-Control-Allow-Credentials' => 'true',
-          'Access-Control-Allow-Methods' => allow_methods,
-          'Access-Control-Max-Age' => '7200'
-        }
-        expect(response.headers).not_to match(hash_including(expected_response))
-      end
-    end
+  it 'should render the CMS page' do
+    get '/'
+
+    expect(response_body).to include(
+      <<~HTML
+        <p>“Yes, I like piña coladas
+        And getting caught in the rain
+        I’m not much into health food
+        I am into champagne
+        I’ve got to meet you by tomorrow noon
+        And cut through all this red tape
+        At a bar called O’Malley’s
+        Where we’ll plan our escape”</p>
+      HTML
+    )
   end
 end
```

Comment:
A lot of these tests seem redundant, or could be simplified. Why not just have one suite of tests with a few edge cases (h1, strong, em, etc), and then run the suite for each route?

Reply:
They're not testing markdown rendering (though they do that too).

They're testing the Rails views now render blobs from the CMS system, and not partials from disk. They also test that the right view is rendering the right CMS content (i.e. `credits.html.haml` should render the the CMS page `/cms/credits` and not `/cms/privacy`)

They're a compatibility shim until these views are deleted when the new client is ready.